code/game/turfs/simulated/floor.dm

```diff
@@ -32,9 +31,31 @@
 	heat_capacity = 10000
 	var/lava = 0
 
+	/// Measures how thick the snow on this turf is.
+	/// Currently this is an arbitrary number; anything higher than 2 isn't meaningful. 0 means there's no snow.
+	/// A turf whose `snow_layers` is `NEVER_HAS_SNOW` cannot accrue snow.
+	/// Check `__defines/turfs.dm` for relevant defines.
+	var/snow_layers = SNOW_NONE
+	/// When characters walk over snowy floors, they leave footprints (see Entered()). This list holds those footprints and gets read when updating the icon.
+	var/list/snow_footprints = list()
+
 /turf/simulated/floor/is_plating()
 	return (!flooring || flooring.is_plating)
 
+/turf/simulated/floor/get_movement_cost()
+	return max(movement_cost, clamp(snow_layers, SNOW_NONE, SNOW_HEAVY))

```

Comment:
right now they serve a different purpose; snow layers can be any arbitrary value, but I want the movement cost to be capped at a certain value to prevent it from being intolerably slow. could go either way though I suppose